src/eei.cpp

```diff
@@ -492,18 +492,25 @@ namespace hera {
       evmc_result call_result;
 
       // Start with base call gas
-      int64_t extra_gas = GasSchedule::call;
+      takeInterfaceGas(GasSchedule::call);
 
+      // These checks are in EIP150 but not in the YellowPaper
       // Charge valuetransfer gas if value is being transferred.
-      // Only charge callNewAccount gas if the account is new and value is being transferred per EIP161.
-      if (!isZeroUint128(call_message.value)) {
-        extra_gas += GasSchedule::valuetransfer;
+      if ((kind == EEICallKind::Call || kind == EEICallKind::CallCode) && !isZeroUint128(call_message.value)) {
```

Comment:
For `CALL`, with `call_message.value == 0`, where the account doesn't already exist, you're not charging `GasSchedule::callNewAccount` here. aleth checks `zeroValueTransferChargesNewAccountGas` which has to do with EIP158 (cf. https://github.com/ethereum/aleth/blob/37ead28874c5adb380d4c60eadff161ca9736ece/libethcore/EVMSchedule.h#L86). Can you confirm that this is intentional?

Reply:
That aleth code is super confusing and I have no idea what each of those random rules mean and on what hard fork they are enabled. We are copying byzantium behaviour. More or less.

Reply:
This is ok. This is byzantium behavior.